File: 2022/js/Day-01/Day-01.js
const fs = require("fs");

inputData = fs.readFileSync("input.txt", "utf8");
inputData = inputData.split(/\r?\n/);
//add an extra empty item cause that what marks "next elf"
inputData.push("");

let elvesCalories = [];

let caloriesToAdd = [];
for (data of inputData) {
  if (data) {
    caloriesToAdd.push(Number(data));
  } else {
    elvesCalories.push(caloriesToAdd.reduce((sum, v) => sum + v));
    caloriesToAdd = [];
  }
}

const Solution01 = (elvesCalories) => {
  return Math.max(...elvesCalories);
};

const Solution02 = (elvesCalories) => {
  elvesCalories = [...elvesCalories];
  elvesCalories = elvesCalories.sort((a, b) => b - a);
  elvesCalories.splice(3);

  return elvesCalories.reduce((sum, v) => sum + v);
};

console.log(`Solution 01: ${Solution01(elvesCalories)}`);
console.log(`Solution 02: ${Solution02(elvesCalories)}`);

File: 2024/python/Day-01/Day_01.py
numbers_a = []
numbers_b = []

with open("input.txt", "r") as f:
    for line in f.readlines():
        a, b = line.strip().split()
        numbers_a.append(int(a))
        numbers_b.append(int(b))


def solution01(numbers_a, numbers_b):
    return sum([abs(a - b) for a, b in zip(sorted(numbers_a), sorted(numbers_b))])


def solution02(numbers_a, numbers_b):
    return sum([a * numbers_b.count(a) for a in numbers_a])


print("—")
print(f"Solution 01: {solution01(numbers_a, numbers_b)}")
print(f"Solution 02: {solution02(numbers_a, numbers_b)}")
print("—")
